cmd/gossamer/genesis.go

```diff
@@ -27,100 +27,114 @@ import (
 	"github.com/ChainSafe/gossamer/lib/database"
 	"github.com/ChainSafe/gossamer/lib/genesis"
 	"github.com/ChainSafe/gossamer/lib/trie"
-	"github.com/ChainSafe/gossamer/lib/utils"
 	"github.com/ChainSafe/gossamer/node/gssmr"
 
 	log "github.com/ChainSafe/log15"
 	"github.com/urfave/cli"
 )
 
-func loadGenesis(ctx *cli.Context) error {
-	currentConfig, err := getConfig(ctx)
+// initializeNode creates a Genesis instance from the configured genesis file
+// and then initializes the state database using the Genesis instance
+func initializeNode(ctx *cli.Context) error {
+	cfg, err := getConfig(ctx)
 	if err != nil {
 		return err
 	}
 
-	// read genesis file
-	genesisPath := getGenesisPath(ctx)
+	datadir := cfg.Global.DataDir
+	genPath := getGenesisPath(ctx)
 
-	dataDir := utils.ExpandDir(currentConfig.Global.DataDir)
-	if ctx.String(DataDirFlag.Name) != "" {
-		dataDir = utils.ExpandDir(ctx.String(DataDirFlag.Name))
-	}
-
-	log.Debug("Loading genesis", "genesisPath", genesisPath, "dataDir", dataDir)
+	log.Info(
+		"[gossamer] Initializing node...",
+		"datadir", datadir,
+		"genesis", genPath,
+	)
 
-	// read genesis configuration file
-	gen, err := genesis.LoadGenesisFromJSON(genesisPath)
+	// load Genesis from genesis configuration file
+	gen, err := genesis.LoadGenesisFromJSON(genPath)
```

Comment:
genPath abbreviated

Reply:
my thinking was that we have `gen` already, so we should use `genPath` instead of `genesisPath`

Reply:
also, I like to keep variables as short and consistent as possible, saving the full name (ie, `genesis`) for the package name (ie, `genesis.Genesis`).